daemon/src/supervisor.rs

```diff
@@ -104,6 +120,29 @@ where
     }
 }
 
+#[async_trait]
+impl<T, R> xtra::Handler<Panicked> for Actor<T, R>
+where
+    T: xtra::Actor,
+    R: fmt::Display + 'static,
+{
+    async fn handle(&mut self, msg: Panicked, ctx: &mut Context<Self>) {
+        let actor = T::name();
+
+        match msg.error.downcast::<String>() {
+            Ok(reason) => {
+                tracing::error!("{actor} panicked: {reason}");
+            }
+            Err(_) => {
+                tracing::error!("{actor} panicked");
+            }
+        }
+
+        self.metrics.num_panics += 1;
+        self.spawn_new(ctx)
```

Comment:
❓ Why is panicking treated as an automatic restart (as opposed to stopping, which depends on the restart policy)?

Reply:
So I figured that because the cause of the panic is unknown at coding time, it doesn't make much sense to implement a policy.

If I would know the problem already to code the policy, I could also just fix the problem in the first place 😁

The idea of this experiment is that hopefully the panic is temporary and restarting the actor puts it in a known state where we can progress again. Perhaps only for a while if the panic was triggered by recurring data.

So we certainly need to monitor these metrics!

I am thinking of rewriting the trace statements here and putting in a dedicated subscriber. Probably in a follow-up though!